app/services/write_offer.rb

```diff
@@ -0,0 +1,28 @@
+class WriteOffer
+  def initialize(conditions:, application_choice:)
+    @conditions = conditions
+    @application_choice = application_choice
+  end
+
+  def call
+    offer = Offer.find_or_initialize_by(application_choice: @application_choice)
+    OfferCondition.where(offer: offer).delete_all
```

Comment:
This leaves some orphaned audits. Should we delete them too?

Reply:
I understand that this is easier to achieve, but I think we should only be deleting condition that no longer exist, and only creating new ones (or ones that have been reworded). That way all audit entries will also be relevant.

Reply:
Yeah, it would be nice to have it do that. It initially felt like more effort than it's worth though.

We can take the list of conditions passed in and match them to existing conditions. Then we are left with some new condition strings, and a list of unmapped existing OfferConditions. We'd have to randomly update the OfferConditions with the new condition strings, and add/delete OfferConditions to match the numbers.

All we are actually trying to achieve here is to keep the state of Offer & OfferCondition consistent with what's in `application_choice.offer` (more like a rolling backfill). When we split up the conditions to be managed individually, we will need to worry about this (in particular preserving condition statuses) and we'll be in a better position to make structural changes to support it.

What do you think?

Reply:
As discussed after the standup, we can leave this like this for now.

Reply:
We've disabled auditing on the OfferConditions model to avoid leaving orphaned audits. We will re-add this in the future